### PR TITLE
Change Payload

### PR DESCRIPTION
**Pull request by dolfelt**
@shadowhand @elazar @ameech 

This is just the initial Payload. I'm wondering how we should handle setting HTTP status codes and other response information given the change in the Payload.

### CODE REVIEW

**Comment by elazar** [comment]
How would redirects be handled with the statuses that are available in this PR?

File: tests/Responder/AbstractResponderTest.php [hunk 1]
```diff
@@ -50,14 +50,13 @@ public function testStatusCode($status, $expected)
     {
         $payload = null;
         if ($status) {
```
Comment (shadowhand) [inline tests/Responder/AbstractResponderTest.php]: The dataprovider for this doesn't really work anymore.

**Comment by dolfelt** [comment]
@elazar @shadowhand That's what I'm struggling with. How general do we want the `Responder` to be? Do we want to basically force every to write their own, or should we provide sane defaults with several different HTTP status codes, redirect ability, etc.?

**Comment by shadowhand** [comment]
I think we want to provide a abstract responder only (or one that only provides a 200 OK response) and a bunch of traits for things like redirect.

**Comment by shadowhand** [comment]
I have some theories about how that would look if you want me to take over the PR.

**Comment by dolfelt** [comment]
@shadowhand Sure, I'm fine with that. I really do think we should provide a sane default though. Like a `BasicResponder` or `JsonResponder` that brings together the abstract and traits. Then projects can extend as necessary, or use the traits.

**Comment by shadowhand** [comment]
Once the abstract responder was simplified, it didn't seem necessary to use traits to implement the other responders. Let me know [what you think of this](https://github.com/sparkphp/Spark/blob/feature/payload-responder/src/Responder/AbstractResponder.php).

**Comment by shadowhand** [comment]
Tests are still broken, but the AbstractResponderTest has been fixed.

**Comment by elazar** [comment]
Seems like `AbstractResponder` in its current form should be named `AbstractHttpResponder`, being that extending it wouldn't really seem appropriate for CLI (e.g. `getHttpStatus()`)?

**Comment by shadowhand** [comment]
@elazar that applies to all of ADR really, except for the domain classes.

**Comment by elazar** [comment]
@shadowhand Well, that begs the question: how are we going to handle instances where we need to do things via command line, if not with Spark?

**Comment by shadowhand** [comment]
@elazar I think we continue to use the domain/payload pattern but use [CLImate](https://github.com/thephpleague/climate) for the I/O.

**Comment by shadowhand** [comment]
Tests are now passing. This is effectively complete.

**Comment by dolfelt** [comment]
:+1:

We ready to merge this in?

_Random thought: If someone wanted to create a Plates responder, are we assuming they would extend the payload to include additional details? Is that preferred?

**Comment by shadowhand** [comment]
@dolfelt Plates responder works with the current Payload (or should, I don't have tests for it) without any modifications.

**Comment by dolfelt** [comment]
Awesome! :shipit:

**Pull request merged**